server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -154,9 +167,74 @@ public RepositoriesService(
             threadPool.relativeTimeInMillisSupplier()
         );
         this.preRestoreChecks = preRestoreChecks;
+        this.defaultRepository = DEFAULT_REPOSITORY_SETTING.get(settings);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(DEFAULT_REPOSITORY_SETTING, this::setDefaultRepository, this::validateDefaultRepository);
         snapshotMetrics.createSnapshotShardsInProgressMetric(this::getShardSnapshotsInProgress);
     }
 
+    /**
+     * Gets the configured default repository.
+     *
+     * @return the default repository name, or an empty string if not configured
+     */
+    public String getDefaultRepository() {
+        return defaultRepository;
+    }
+
+    /**
+     * Sets the default repository value.
+     * This is called by the settings update consumer when the setting changes.
+     *
+     * @param repositoryName the new default repository name
+     */
+    private void setDefaultRepository(String repositoryName) {
+        this.defaultRepository = repositoryName;
+        logger.info("Default repository set to [{}]", repositoryName);
+    }
+
+    /**
+     * Validates that the default repository is registered and not read-only.
+     * This validator is called when the default repository setting is updated.
+     * Empty strings are allowed to clear the default repository setting.
+     *
+     * Only runs on master node/master update thread to avoid issues at node start time
+     *
+     * @param repositoryName the repository name to validate
+     * @throws IllegalArgumentException if the repository name is not empty and the repository doesn't exist or is not empty and read-only
+     */
+    private void validateDefaultRepository(String repositoryName) {
+        // Only validate on the master update thread to avoid validation issues at node start time
+        // when repositories may not be initialized yet.
+        if (MasterService.isMasterUpdateThread() == false) {
+            logger.debug("Skipping validation of default repository [{}] outside master update thread", repositoryName);
+            return;
+        }
+        if (Strings.isEmpty(repositoryName)) {
+            logger.info("Default repository cleared");
+            return;
+        }
+        Repository repository = repositoryOrNull(ProjectId.DEFAULT, repositoryName);
```

Comment:
`repositoryOrNull` checks `internalRepositories` as well. We don't want that since internal repositories are for CCR and cannot be used for anything else.